### DISCUSSION TITLE
Dynamic subtype check

### DISCUSSION THREAD

**Discussion by eaubin** [pydantic/pydantic#9493] [Question]
Do you have any suggestions for implementing a dynamic check that one model is a subtype of another? Perhaps a plugin or api I can't find? I have an operation f: X->Y and cs:collection[X'] and I want to check X'<:X before calling f on each item, but do not want to validate each item in the collection.

**Comment by uriyyo** [comment]
Hi @eaubin,

You can try to use `InstanceOf` validator:
```py
from pydantic import InstanceOf

collection: list[InstanceOf[X]]
```

More info - https://docs.pydantic.dev/latest/api/functional_validators/#pydantic.functional_validators.InstanceOf

**Comment by eaubin** [comment]
Thanks @uriyyo I'm looking for something a bit different though, I'm really interested in the dynamic subtype checking of the model types, not the instances.  A different example use case might be defining a composition function that fails at composition time rather than application time when incompatible return and parameter types are used. 

```python
def compose(f, g, f_return_type: Type[BaseModel], g_parameter_type: Type[BaseModel]):
    if not some_unknown_function_here_to_check_subtyping(f_return_type, g_parameter_type):
        raise ValueError('f not compatible with g')
    else:
        return lambda(x: g(f(x)))
```

**Comment by mikenerone** [comment]
I've opened #10462 about this.
